aws/resource_aws_lex_slot_type.go

```diff
@@ -0,0 +1,301 @@
+	d.Set("description", resp.Description)
+	d.Set("last_updated_date", resp.LastUpdatedDate.Format(time.RFC3339))
+	d.Set("name", resp.Name)
+	d.Set("value_selection_strategy", resp.ValueSelectionStrategy)
+
+	if resp.EnumerationValues != nil {
+		d.Set("enumeration_value", flattenLexEnumerationValues(resp.EnumerationValues))
+	}
+
+	version, err := getLatestLexSlotTypeVersion(conn, &lexmodelbuildingservice.GetSlotTypeVersionsInput{
+		Name: aws.String(d.Id()),
+	})
+	if err != nil {
+		return err
+	}
+	d.Set("version", version)
+
+	return nil
+}
+
+func getLatestLexSlotTypeVersion(conn *lexmodelbuildingservice.LexModelBuildingService, input *lexmodelbuildingservice.GetSlotTypeVersionsInput) (string, error) {
+	version := "$LATEST"
+
+	for {
+		page, err := conn.GetSlotTypeVersions(input)
+		if err != nil {
+			return "", err
+		}
+
+		// At least 1 version will always be returned.
+		if len(page.SlotTypes) == 1 {
+			break
+		}
+
+		for _, slotType := range page.SlotTypes {
+			if *slotType.Version == LexSlotTypeVersionLatest {
+				continue
+			}
+			if *slotType.Version > version {
+				version = *slotType.Version
+			}
+		}
+
+		if page.NextToken == nil {
+			break
+		}
+		input.NextToken = page.NextToken
+	}
+
+	return version, nil
+}
+
+func resourceAwsLexSlotTypeUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).lexmodelconn
+
+	input := &lexmodelbuildingservice.PutSlotTypeInput{
+		Checksum:               aws.String(d.Get("checksum").(string)),
+		CreateVersion:          aws.Bool(d.Get("create_version").(bool)),
+		Description:            aws.String(d.Get("description").(string)),
+		Name:                   aws.String(d.Id()),
+		ValueSelectionStrategy: aws.String(d.Get("value_selection_strategy").(string)),
+	}
+
+	if v, ok := d.GetOk("enumeration_value"); ok {
+		input.EnumerationValues = expandLexEnumerationValues(v.(*schema.Set).List())
+	}
+
+	err := resource.Retry(d.Timeout(schema.TimeoutUpdate), func() *resource.RetryError {
+		_, err := conn.PutSlotType(input)
+
+		if isAWSErr(err, lexmodelbuildingservice.ErrCodeConflictException, "") {
+			return resource.RetryableError(fmt.Errorf("%q: slot type still updating", d.Id()))
+		}
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("error updating slot type %s: %w", d.Id(), err)
+	}
+
+	return resourceAwsLexSlotTypeRead(d, meta)
+}
+
+func resourceAwsLexSlotTypeDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).lexmodelconn
+
+	err := resource.Retry(d.Timeout(schema.TimeoutDelete), func() *resource.RetryError {
+		_, err := conn.DeleteSlotType(&lexmodelbuildingservice.DeleteSlotTypeInput{
+			Name: aws.String(d.Id()),
+		})
+
+		if isAWSErr(err, lexmodelbuildingservice.ErrCodeConflictException, "") {
+			return resource.RetryableError(fmt.Errorf("%q: slot type still deleting", d.Id()))
```

Comment:
Are you seeing an error deleting? I have never seen an issue here. See my comments below about the versions not being able to exist without the slot type itself.

Reply:
Yes, because there is a slight delay in deletion, especially when there are multiple versions, I was occasionally hitting errors. The two functions described above are a standard pattern that we've adopted for handling this.